test/jdk/java/awt/Frame/GetBoundsResizeTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 1998, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import java.awt.Button;
+import java.awt.Frame;
+import java.awt.EventQueue;
+import java.awt.Robot;
+
+/*
+ * @test
+ * @bug 4103095
+ * @summary Test for getBounds() after a Frame resize.
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual GetBoundsResizeTest
+*/
+
+public class GetBoundsResizeTest {
+    private static final String INSTRUCTIONS = """
+            0. There is a test window with a "Press" button,
+                Its original bounds should be printed in the text area below.
+            1. Resize the test window using the upper left corner.
+            2. Press the button to print the result of getBounds() to the terminal.
+            3. If getBounds() prints the correct values for the window, click Pass,
+             otherwise click Fail.
+            """;
+
+    private static JTextArea textArea;
+    private static Frame frame;
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+
+        PassFailJFrame passFailJFrame = PassFailJFrame
+                .builder()
+                .title("GetBoundsResizeTest Instructions")
+                .instructions(INSTRUCTIONS)
+                .splitUIBottom(() -> {
+                    textArea = new JTextArea("", 8, 55);
+                    textArea.setEditable(false);
+                    return new JScrollPane(textArea);
+                })
+                .testUI(GetBoundsResizeTest::getFrame)
+                .rows(7)
+                .columns(40)
+                .build();
+
+        robot.waitForIdle();
+        robot.delay(500);
+
+        EventQueue.invokeAndWait(() ->
+                textArea.append("Original Frame.getBounds() = %s\n"
+                        .formatted(frame.getBounds())));
```

Comment:
The same applies to the following event handler.

Reply:
What can help to reduce the indentation is introducing a helper method. If it's still too long, store the result of formatting into a local variable and then append it to `textArea`. This can be done even in lambdas.

Reply:
@aivanov-jdk oh I might have missed the chaining order. Thanks for catching it.